### PR TITLE
front: add settings for VITE warnings & errors overlay

### LINKED ISSUE #3567: front: add settings for VITE warnings & errors overlay

### PR DESCRIPTION
**Pull request by nicolaswurtz**
closes #3567

### CODE REVIEW

File: front/.env.local.defaults [hunk 1]
```diff
@@ -5,3 +5,6 @@ OSRD_KEYCLOAK_REALM=
 OSRD_SENTRY_DSN=
 OSRD_SENTRY_ENVIRONMENT=
 OSRD_GIT_DESCRIBE=
+OSRD_VITE_PORT=3000
+OSRD_VITE_OVERLAY=true
+OSRD_VITE_OVERLAY_IS_OPEN=false
```
Comment (anisometropie) [inline front/.env.local.defaults]:
```suggestion
OSRD_VITE_OVERLAY_OPEN_BY_DEFAULT=false
```

**Review by anisometropie — APPROVED** [review]

File: front/vite.config.ts [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
+        example: '.env.example',
+      }),
+      checker({
+        typescript: {
+          buildMode: true,
+        },
+        eslint: {
+          lintCommand: 'eslint --ext .ts,.tsx,.js,.jsx src --max-warnings 203',
+        },
+        overlay: env.OSRD_VITE_OVERLAY !== 'false' && {
```
Comment (anisometropie) [inline front/vite.config.ts:27]:
```suggestion
        overlay: env.OSRD_VITE_OVERLAY === 'true' && {
```

`env.OSRD_VITE_OVERLAY !== 'false'` is true for any string except `'false'`

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/.env.local.defaults (hunk 1 above)
Comment (nicolaswurtz) [reply front/.env.local.defaults]: you're right, done !

File: front/vite.config.ts (hunk 2 above)
Comment (nicolaswurtz) [reply front/vite.config.ts:27]: Yes and it's exactly what we want : if not assigned, it's true by default.

**Review by anisometropie — DISMISSED** [review]

**Review by anisometropie — APPROVED** [review]

**Pull request merged**